.github/workflows/test.yml

```diff
@@ -0,0 +1,36 @@
+name: Run unit and ui tests
+
+on:
+  push:
+    branches-ignore:
+      - 'develop'
+      - 'main'
+  
+jobs:
+  test:
+    runs-on: macos-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup Ruby 3.2.1
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2.1'
+          bundler-cache: true
+      
+      - name: Netrc api.github.com
+        uses: extractions/netrc@v1
+        with:
+          machine: maven.pkg.github.com
+          username: ${{ secrets.NETRC_USERNAME }}
+          password: ${{ secrets.NETRC_TOKEN }}
+    
+      - name: Setup xcodegen and init xcode project
+        uses: xavierLowmiller/xcodegen-action@1.1.2
+        with:
+          spec: project.yml
+          use-cache: true
+
+      - name: Run tests with fastlane
+        run: bundle exec fastlane test
```

Comment:
This should now also be added to the publish workflow (right before starting the release)

Reply:
will do in separate PR